x-pack/plugins/security/server/authorization/validate_es_response.ts

```diff
@@ -46,7 +49,7 @@ function buildValidationSchema(application: string, actions: string[], resources
         }
 
         Object.values(value).forEach((actionResult) => {
-          actionValidationSchema.validate(actionResult);
+          anyBoolean.validate(actionResult);
         });
```

Comment:
__note:__ I initially thought that we could also have something like `anyBooleanArray = schema.arrayOf(schema.boolean())` (and `schema.array(schema.object)` for `actionsValidationSchema`) to squeeze a bit more. But it turned out that the benefit is insignificant with the data size we have, wanted to mention anyway though. Here are few test cases:
```ts
const anyBoolean = schema.boolean();
const anyBooleanArray = schema.arrayOf(schema.boolean());
const testFixture = Array.from({ length: 99999 }).map((_) => true);

console.time('one-by-one');
testFixture.forEach((actionResult) => anyBoolean.validate(actionResult));
console.timeEnd('one-by-one');

console.time('as-array');
anyBooleanArray.validate(testFixture);
console.timeEnd('as-array');
--------------
one-by-one: 148 ms
as-array: 82 ms
```

But the largest object I've seen while testing this PR has just `498` boolean props (and absolute majority has up to 5 properties):

```ts
const anyBoolean = schema.boolean();
const anyBooleanArray = schema.arrayOf(schema.boolean());
const testFixture = Array.from({ length: 500 }).map((_) => true);

console.time('one-by-one');
testFixture.forEach((actionResult) => anyBoolean.validate(actionResult));
console.timeEnd('one-by-one');

console.time('as-array');
anyBooleanArray.validate(testFixture);
console.timeEnd('as-array');
--------------
one-by-one: 3 ms
as-array: 2 ms
```

Reply:
Ah good call! I'll make this change, every ms counts for this hot path